### ISSUE TITLE
Application Error on Heroku

### ISSUE THREAD

**Issue by JenWardell** [sahat/hackathon-starter#391]
Hi there, 

I've followed the steps in the readme and done the heroku deploy and am encountering some problems. It states application error when deployed on heroku, but runs fine locally. I have checked the mongodb and it is updated to not simply run on localhost. I've checked the .env is named correctly. Is there anything else I might be missing? Is there anything you can suggest? I've tried fresh installs, different versions of node, nothing is working.

**Comment by sahat** [comment]
The first that comes to mind is invalid Mongodb URI. I can't think of anything else from the top of my head. If you have installed [Heroku Toolbelt](https://toolbelt.heroku.com/), you can run `heroku logs` from your app directory to pull the error logs. That will tell you exactly why the app is failing.

**Comment by sahat** [comment]
Oh, did you create environment variables from the Heroku Dashboard? If you have renamed `.env.example` to `.env`, it will not be pushed to GitHub or Heroku. It is "git ignored" on purpose to avoid leaking private keys and password.

You don't have to fill out every env var, but it has to be set, at least to an empty string. If it is not set, it will probably throw and exception somewhere. I know Passport.js strategies don't like it when client id / client secret are undefined.

![sshot-1](https://cloud.githubusercontent.com/assets/544954/13474842/dc06bc6a-e08d-11e5-90d9-1b42d9796945.png)

**Comment by JenWardell** [comment]
I think that might be it, I had manually pushed them but if I didn't do that step it likely wouldn't have caught it. I'll have a look in an hour or two when I'm on my home computer. Thanks for this! Will update when I have tried.

**Comment by JenWardell** [comment]
Hi there, 

So it turned out to be completely my fault. I did the heroku setup with the auto deploy through mongolabs. But I didn't then add MONGODB in the env variables as well, I had just left it as the generated Mongolabs one. Once I realized that and entered it it worked perfectly. Maybe this reminder could be added to the steps in the heroku deploy to save others making silly mistakes like me? Haha. Thanks for the help!
